assets/js/components/surveys/CurrentSurvey.stories.js

```diff
@@ -16,128 +16,106 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import fetchMock from 'fetch-mock';
+
 /**
  * Internal dependencies
  */
-import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
 import WithRegistrySetup from '../../../../tests/js/WithRegistrySetup';
-import CurrentSurvey from './CurrentSurvey';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
 import { CORE_FORMS } from '../../googlesitekit/datastore/forms/constants';
-import * as fixtures from './__fixtures__';
-import fetchMock from 'fetch-mock';
+import CurrentSurvey from './CurrentSurvey';
+import {
+	multiQuestionConditionalSurvey,
+	multiQuestionSurvey,
+	singleQuestionSurvey,
+	singleQuestionSurveyWithNoFollowUp,
+} from './__fixtures__';
 
-const Template = ( { setupRegistry, ...args } ) => (
-	<WithRegistrySetup func={ setupRegistry }>
-		<CurrentSurvey { ...args } />
-	</WithRegistrySetup>
-);
+function Template( { setupRegistry, ...args } ) {
+	return (
+		<WithRegistrySetup func={ setupRegistry }>
+			<CurrentSurvey { ...args } />
+		</WithRegistrySetup>
+	);
+}
+
+function TemplateArgs( data, setupRegistry = () => {} ) {
+	return {
+		setupRegistry( registry ) {
+			fetchMock.post(
+				/google-site-kit\/v1\/core\/user\/data\/survey-event/,
+				{
+					body: {},
+				}
+			);
+
+			const triggerID = 'storybookSurvey';
+
+			const {
+				receiveTriggerSurvey,
+				receiveGetTracking,
+			} = registry.dispatch( CORE_USER );
+
+			receiveTriggerSurvey( data, { triggerID } );
+			receiveGetTracking( { enabled: true } );
+
+			setupRegistry( registry );
+		},
+	};
+}
 
 export const SurveySingleQuestionStory = Template.bind( {} );
 SurveySingleQuestionStory.storyName = 'Single question';
-SurveySingleQuestionStory.args = {
-	setupRegistry: ( registry ) => {
-		fetchMock.post( /google-site-kit\/v1\/core\/user\/data\/survey-event/, {
-			body: {},
-			status: 200,
-		} );
-
-		registry
-			.dispatch( CORE_USER )
-			.receiveTriggerSurvey( fixtures.singleQuestionSurvey, {
-				triggerID: 'storybookSurvey',
-			} );
-		registry.dispatch( CORE_USER ).receiveGetTracking( { enabled: true } );
-	},
-};
+SurveySingleQuestionStory.args = TemplateArgs( singleQuestionSurvey );
```

Comment:
Per my last comment, this could look like so:

```suggestion
SurveySingleQuestionStory.args = {
	setupRegistry( registry ) {
		provideCurrentSurvey( registry, singleQuestionSurvey );
	},
}
```

While this may result in a few more lines, it's easier to follow and extend.

Reply:
Updated.